### ISSUE TITLE
snmptrap V3 username issue

### ISSUE THREAD

**Issue by alfletch** [splunk/splunk-connect-for-snmp#930]
I have an issue where I may be using the same username for multiple secrets, but only the last listed in the values file is used for authentication. 

For example, I have setup secrets 
secret1 -  username=myuser, authkey=myauthkey1, privkey=myprivkey1
secret2 -  username=myuser, authkey=myauthkey1, privkey=myprivkey2

Then in the values.yaml file,
traps:
  usernameSecrets:
     - secret1
     - secret2

In this case when sending traps, only secret2 is authenticated, if I reverse the order of the secrets, then the only traps with secret1 are authenticated.

**Comment by ajasnosz** [comment]
Hello, 
thank you for spotting that, I checked this configuration and it is a bug, that occurs when you have the same usernames. We will try to investigate it further.

**Comment by ajasnosz** [comment]
Yes, from what we tested for now it looked like the issue might be in how pysnmp library is handling the secrets, we will look further into that.

**Comment by ajasnosz** [comment]
Hello All, after looking into pysnmp library this behaviour is the expected one. Pysnmp allows only one-to-one mapping between userName and securityName, which is the key for getting secret configuration. This solution is based on snmp standard mentioned in rfc: https://www.rfc-editor.org/rfc/rfc3414#page-12

**Comment by hassansalama20** [comment]
Hello Team,

In a large heterogeneous environment — for example, above 1000 devices across multiple administrative domains — the current approach (1.16.0/microk8s version) has the following limitations:

Scale: Every secret is registered against every known engine ID. With S secrets and E engine IDs, the USM table grows as S × E. As devices with diverse engine IDs send their first trap, this matrix expands unboundedly.

Username collision: PySNMP's USM table is keyed on (userName, securityEngineId). If two devices use the same SNMPv3 username but different passwords — a common situation across separate administrative domains or vendors — only one entry can exist per (userName, engineId) pair. The second registration silently overwrites the first, causing authentication failures that are difficult to diagnose.

No source-based isolation: There is no mechanism to bind a particular credential to a source IP, subnet, or inventory group. All senders are authenticated against the global credential pool.

No safe multi-tenant support: Different customer segments, network domains, or device vendors cannot maintain independent SNMPv3 credential sets without risk of username collision.

Regards,
Hassan Salama
